source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs

```diff
@@ -28,9 +31,11 @@ public KubernetesContextScriptWrapper(CalamariVariableDictionary variables)
         /// <summary>
         /// One of these fields must be present for a k8s step
         /// </summary>
-        public bool Enabled => !string.IsNullOrEmpty(variables.Get(SpecialVariables.ClusterUrl, "")) ||
+        public bool IsEnabled(ScriptSyntax syntax) => !string.IsNullOrEmpty(variables.Get(SpecialVariables.ClusterUrl, "")) ||
                                !string.IsNullOrEmpty(variables.Get(SpecialVariables.AksClusterName, "")) ||
-                               !string.IsNullOrEmpty(variables.Get(SpecialVariables.EksClusterName, ""));
+                               !string.IsNullOrEmpty(variables.Get(SpecialVariables.EksClusterName, "")) &&
+                                supportedScriptSyntax.Contains(syntax);
```

Comment:
This needs to have `()` around the `!string.IsNullOrEmpty` calls otherwise if either `ClusterUrl` or `AksClusterName` is set and the `syntax` is not supported it will return `true`. E.g.

```
(!string.IsNullOrEmpty(variables.Get(SpecialVariables.ClusterUrl, "")) ||
    !string.IsNullOrEmpty(variables.Get(SpecialVariables.AksClusterName, "")) ||
    !string.IsNullOrEmpty(variables.Get(SpecialVariables.EksClusterName, ""))) &&
    supportedScriptSyntax.Contains(syntax)
```